aiobotocore/eventstream.py

```diff
@@ -0,0 +1,54 @@
+from botocore.eventstream import EventStream, EventStreamBuffer
+
+
+class AioEventStream(EventStream):
+    def __init__(self, *args, **kwargs):
+        super(AioEventStream, self).__init__(*args, **kwargs)
+        self._buffer = EventStreamBuffer()
+
+    def _create_raw_event_generator(self):
```

Comment:
I think you can use the async_generator module I added to make this more similar to the sync version, see paginate.py for usage

Reply:
I think then you won't need the `__init__` method and maybe simplifies `__anext__`

Reply:
Oh yeah, that's genius. I'll have a look this evening as I reckon your right and it should simplify everything. All of that was purely as I couldn't use `yield` inside an async function :D